display feed icons on frontpage

Niehztog opened the pull request:
This PR adds the display of the icon of each bridge on the front page of rss bridge. If the icon uri cannot be resolved, the icon will be hidden.
The code change in `FrontpageAction.php` is just a small refactoring and is optional.

Niehztog committed:
display feed icons on frontpage

Niehztog committed:
adds feed icons for BlizzardNewsBridge and NiusBridge

gsantner commented:
This is a really bad change I would say.

It means that if the provider enabled lets say 100 bridges that has icon defined - then each visitor of frontpage makes additional 100 HTTP requests to bridge **!!external!! URLs**. Which **can be abused for tracking users of rss-bridge** on each of these 100 sites without visitors knowing it. Without using the bridge even.

If this has to be done really at dashboard, I would suggest to make it so rss-bridge fetches the icon on it's own and the image is served from rss-bridge **cache only** .

Just by example of this PR, it adds a quite unique URL which looks concerning.

> b34a003c6f2f637ee8f4f7b406f3b9b120b918c04cabec7f03a760e708977ea9689a1c638f4396def8dce7b202cd007eae91946cc3c4a578aa8b5694226cfc6.ico

![grafik](https://github.com/RSS-Bridge/rss-bridge/assets/6735650/5671a82f-1f70-4993-8c1f-7acccc985773)

dvikan commented:
i agree. resource heave and privacy issue.

rssbridge could act as a proxy and serve icons but that would be a new feature.

Niehztog committed:
adds image proxy script for caching and increased security when displaying feed icons

Niehztog commented:
Hello @dvikan and @gsantner ,
thank you very much for your valuable feedback regarding security/privacy and ressource heave of my PR. I addressed all those issues in my latest commit. Please take a look.

Niehztog committed:
prevent repeated loading attempts of broken feed icons
apply timeout of 10 seconds when loading feed icons

Niehztog committed:
fix linter errors

Niehztog commented:
The code has been further improved to decrease loading times and save performance. This has been accomplished by applying a timeout value of 10 seconds when loading images and storing information about broken images in cache to prevent continued attempts to load them.

The PR is now ready of being merged.

Niehztog committed:
decrease timeout value, specify custom user agent

dvikan commented:
@Niehztog thanks for the effort but wont merge this

Niehztog commented:
@dvikan could you please explain your decision? All issues are addressed, so whats the problem?

dvikan commented:
Main reason is visiting the front page produces 438 network calls which is too burdensome/slow/annoying for users.

Also low code quality. Sorry.

Niehztog commented:
@dvikan Thank you for clarification. We could work together to address those issues you see in the code quality. You could just tell me which parts need to be improved.
The issue with the amount of network requests could be addressed by limiting display of feed icons only to activated bridges.

dvikan commented:
For example on https://rss-bridge.org/bridge01/ all 436 bridges are enabled/activated. We cant generate that many network calls for a single user. Not good ux.

Niehztog commented:
I agree
